src/App.tsx

```diff
@@ -1,14 +1,44 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useLayoutEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { ITodoFilter, Todo } from './types/Todo';
+import { getTodos } from './api';
+import { getFiltredTodo } from './utils/filterTodo';
 
 export const App: React.FC = () => {
+  const [isLoading, setLoading] = useState(true);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setselectedTodo] = useState<null | Todo>(null);
+  const [todosFilter, setTodosFilter] = useState<ITodoFilter>({
+    status: '',
```

Comment:
u can create `enum` for ur statuses

```suggestion
    status: '',
```
```
export enum Filter_Statuses {
  All = 'all',
  Active = 'active',
  Completed = 'completed',
}
```